include/RAJA/index/RangeSegment.hpp

```diff
@@ -69,8 +69,18 @@ namespace RAJA
  *
  ******************************************************************************
  */
-template <typename StorageT, typename DiffT = strip_index_type_t<StorageT>>
+
+template<typename T>
+using make_signed_t = typename std::conditional < 
```

Comment:
Is the conditional needed? Also, this should be moved to the same file as `strip_index_type_t`

Reply:
conditional is needed in order to allow for the use of floating types. std::make_signed cannot natively handle floats as it is only designed to be used with integral types. However a case could be made that floating types should not be allowed to be used with RangeSegments as they are not compatible with RangeStrideSegment objects.

Reply:
Could we still move this to a more centralized place?